### PR TITLE
refactor: preserve app state

### PR DESCRIPTION
**Pull request by alexcasche**
**What This PR Does**
- fixes bug where updating config resets appearance of all content models using the nacelle app
- updates packages to expose required method `sdk.app.getCurrentState`

**How To Test**
- clone the repo
- `yarn start`
- open Prarie Wind Apparel experimental space
- click on `Apps > Manage Apps` and click on `Nacelle [Local Dev]`
- update the endpoint from `v2` to `v3` or vice versa
- open up `Content Models > Collection Grid` and click on Setting next to `Collection Handle`
- ensure that the field's appearance is still set to `Nacelle [Local Dev]`

### CODE REVIEW

File: .eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...":"26"},{"filePath":"45","messages":"46","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"26"},"/Users/alexasche/Desktop/repos/contentful-app/src/index.tsx",[],["47","48"],"/Users/alexasche/Desktop/repos/contentful-app/src/components/ConfigScreen.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/components/Dialog.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/components/Page.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/components/Field.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/components/EntryEditor.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/components/Sidebar.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/components/Paginator.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/components/ResourceListItem.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/queries/hail-frequency.tsx",[],"/Users/alexasche/Desktop/repos/contentful-app/src/queries/warp2.tsx",[],{"ruleId":"49","replacedBy":"50"},{"ruleId":"51","replacedBy":"52"},"no-native-reassign",["53"],"no-negated-in-lhs",["54"],"no-global-assign","no-unsafe-negation"]
```
Comment (NWRichmond) [inline .eslintcache:1]: I think we need to remove this file from version control and add `.eslintcache` to `.gitignore`.

**Review by NWRichmond — CHANGES_REQUESTED** [review]
Awesome! Thanks for those testing steps - everything worked as expected:

https://user-images.githubusercontent.com/5732000/149549399-34094310-a165-4187-b5db-4f10161fb4b7.mov

Before merging, let's just get rid of the `.eslintcache` file and add it to `.gitignore`.

**Comment by alexcasche** [comment]
@NWRichmond should be all set now!  Thinking this release would be a patch?

**Review by NWRichmond — APPROVED** [review]
Thanks for those changes, @alexcasche!

> Do we have a deploy process for this?

It's just hosted on Netlify (not Heroku / some other container) so I think we're all set!

**Pull request merged**